src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/ICastable.cs

```diff
@@ -2,16 +2,15 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-//
-// Support for dynamic interface casting. Specifically implementing this interface on a type will allow the
-// type to support interfaces (for the purposes of casting and interface dispatch) that do not appear in its
-// interface map.
-//
-
 using System;
 
 namespace System.Runtime.CompilerServices
 {
+    /// <summary>
+    /// Support for dynamic interface casting. Specifically implementing this interface on a type will allow the
+    /// type to support interfaces (for the purposes of casting and interface dispatch) that do not appear in its
+    /// interface map.
+    /// </summary>
```

Comment:
Does it really make sense to unify this runtime implementation detail? Any sort of change to this interface will have a lot of consequences on the runtime and compiler, and whatever class implements this interface - it can't just mirror. The comments we left around after `CastableObject` work seem to also indicate we would actually like to kill this type in CoreRT:

https://github.com/dotnet/corert/blob/8e7e4b5e070e95fed154706511a022433ff342b7/src/Runtime.Base/src/System/Runtime/TypeCast.cs#L290

Reply:
This interface is same between CoreCLR, CoreRT and Mono today. We can always undo this sharing if it becomes a problem.